src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -2153,58 +2153,58 @@ internal static int IndexOfAnyValueType<T>(ref T searchSpace, T value0, T value1
             else if (Vector256.IsHardwareAccelerated && length >= Vector256<TValue>.Count)
             {
                 Vector256<TValue> equals, values = Vector256.Create(value);
-                ref TValue currentSearchSpace = ref Unsafe.Add(ref searchSpace, length - Vector256<TValue>.Count);
+                int offset = length - Vector256<TValue>.Count;
 
                 // Loop until either we've finished all elements or there's less than a vector's-worth remaining.
                 do
                 {
-                    equals = TNegator.NegateIfNeeded(Vector256.Equals(values, Vector256.LoadUnsafe(ref currentSearchSpace)));
+                    equals = TNegator.NegateIfNeeded(Vector256.Equals(values, Vector256.LoadUnsafe(ref searchSpace, (nuint)offset)));
```

Comment:
`(nuint)offset` will be a sign-extending move, so less ideal.
Either
```suggestion
                    equals = TNegator.NegateIfNeeded(Vector256.Equals(values, Vector256.LoadUnsafe(ref searchSpace, (uint)offset)));
```
or declare as `nint offset` (then the cast is just a `mov` instead of `movsxd`).